src/main/java/com/amplitude/HttpCall.java

```diff
@@ -14,21 +14,35 @@
 
 import java.util.List;
 
-public class HttpCall {
+enum HttpCallMode {
+  REGULAR_HTTPCALL,
+  BATCH_HTTPCALL
+};
 
+public abstract class HttpCall {
   /*
    * Use HTTPUrlConnection object to make async HTTP request,
    * using data from event like device, class name, event props, etc.
    *
    * @return The response object which contains a code and other information
    */
-  protected static Response syncHttpCallWithEventsBuffer(List<Event> events, String apiKey) {
+  public String apiKey;
+  public String apiUrl;
+
+  protected HttpCall(String apiKey, String apiUrl) {
```

Comment:
one possible further way we can consider is in HttpCallMode, each enum will hold the url value, then HttpCall just take HttpCallMode as constructor. This way it can force the HttpCall to only use limited url. As long as HttpCallMode enum is not exposed to public that is fine for me.

Reply:
Another thing we can consider is this constructor just take an apiKey, then in abstract class we just have an abstract method getApiUrl, which returns a string. Then we don't need to think about the last comment. And in each sub class, we can just overwrite the method getApiUrl.

Reply:
I Will choose the second suggestion since we can take advantage of the abstract class and make the constructor simpler.